### PR TITLE
fix(slow-start-up): refactors worker/modv lifecycle and module loading

### LINKED ISSUE #617: modV start-up is slow
**Operating system and application version**
 - OS: All
 - Version 3.18.0

**Describe the bug**
Since #598, modV has many more modules to load at start-up.
When modules are loaded, the gallery UI struggles to keep up which slows down the initial startup as the UI is essentially blocking other UI updates. 

**To Reproduce**
1. Open modV

**Expected behavior**
modV's UI does not slow down the start-up process.

**Screenshots**
n/a

**Additional context**
Initial suggested fix:

* implement a splash screen, hide the main window until all modules are loaded, refactor so the UI is mounted after the worker is done loading modules and then show the main window, hide the splash screen.

### PR DESCRIPTION
**Pull request by 2xAA**
The worker now reports to the renderer process when it's done loading. main.js is waiting for modV's ready promise to resolve before mounting the Vue application. modV also sends an event to the background process to show the main window and close the splash screen.

Speed gains can be measured on macOS (Intel). Windows seems about the same, but at least the UI isn't unresponsive while it's loading.

fixes #617

### CODE REVIEW

**Comment by TimPietrusky** [comment]
@2xAA this looks super cool, thank you very much!

I just have one error:

1. Open modV
2. Reload modV
3. See the following error in a popup:

```js
Uncaught Exception:
TypeError: Object has been destroyed
at IpcMainImpl.eval (webpack:///./src/background.js?:123:16)
at IpcMainImpl.emit (events.js:327:22)
at Object.<anonymous> (electron/js2c/browser_init.js:157:9051)
at Object.emit (events.js:315:20)
```

This happens every time modV is reloaded.

**Comment by 2xAA** [comment]
I actually want to hold off on this until #623 is merged as the queuing of messages improves start-up time drastically.
We can still merge this to have a nice splash-screen, once that error is fixed.

File: src/subpages/splash-screen/main.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import Vue from "vue";
+import App from "./App.vue";
+
+debugger;
```
Comment (TimPietrusky) [inline src/subpages/splash-screen/main.js]: Can you please get rid of the `debugger`?

**Comment by TimPietrusky** [comment]
@2xAA thanks for the fix. Now when I reload modV, the whole window is white (excluding the navigation) until everything is loaded. Maybe add a loading spinner to the main window background? So that it's indicating that the application is loading? 

And also could you please resolve the conflicts? 🙏

**Review by TimPietrusky — CHANGES_REQUESTED** [review]

File: src/subpages/splash-screen/main.js (hunk 1 above)
Comment (TimPietrusky) [reply src/subpages/splash-screen/main.js]: This was resolved!

**Comment by TimPietrusky** [comment]
@2xAA 

* Set background to default background color
* Added loading spinner to improve the waiting experience

What do you think?

**Comment by 2xAA** [comment]
Screenshot 2021-09-22 at 10 58 02

@TimPietrusky looks good, though the splash screen now has a background and loading animation 😅

**Comment by TimPietrusky** [comment]
@2xAA the issues have been addressed, please take another look!

**Comment by 2xAA** [comment]
@TimPietrusky Awesome, seems to work fine :)

**Pull request merged**